src/interface.rs

```diff
@@ -113,33 +110,41 @@ enum Error {
     RangeError,
 }
 
-fn evaluate_function_over_domain(start_x: f64, end_x: f64, resolution: u32, function_string: &str) -> Result<Vec<(f64, f64)>, Error> {
+fn evaluate_function_over_domain(
```

Comment:
Could you possibly refactor this function as part of the rearchitecture?

This function makes the most sense to be a publicly exported function in the `sexe_expression` crate. We should also change this from taking `function_string` to an actual expression node. This means that where we call this function, we will need to first parse the function string, then pass the expression node to _this_ function in the expression module.